src/runtime_src/core/common/query_requests.h

```diff
@@ -1003,6 +1012,87 @@ struct xmc_reg_base : request
   get(const device*) const = 0;
 };
 
+struct xmc_scaling_support : request
+{
+  using result_type = bool;       // get value type
+  static const key_type key = key_type::xmc_scaling_support;
+
+  virtual boost::any
+  get(const device*) const = 0;
+};
+
+struct xmc_scaling_critical_temp_threshold : request
+{
+  using result_type = std::string;       // get value type
+  static const key_type key = key_type::xmc_scaling_critical_temp_threshold;
+
+  virtual boost::any
+  get(const device*) const = 0;
+};
+
+struct xmc_scaling_critical_pow_threshold : request
+{
+  using result_type = std::string;       // get value type
+  static const key_type key = key_type::xmc_scaling_critical_pow_threshold;
+
+  virtual boost::any
+  get(const device*) const = 0;
+};
+
+struct xmc_scaling_threshold_power_limit : request
+{
+  using result_type = std::string;       // get value type
+  static const key_type key = key_type::xmc_scaling_threshold_power_limit;
+
+  virtual boost::any
+  get(const device*) const = 0;
+};
+
+struct xmc_scaling_threshold_temp_limit : request
+{
+  using result_type = std::string;       // get value type
+  static const key_type key = key_type::xmc_scaling_threshold_temp_limit;
+
+  virtual boost::any
+  get(const device*) const = 0;
+};
+
+struct xmc_scaling_threshold_power_override_en : request
```

Comment:
If enable is true/false, please change the return type to bool. Same for the following function. We have some examples in this file for reference